api/src/main/java/io/opentelemetry/OpenTelemetry.java

```diff
@@ -5,255 +5,194 @@
 import io.opentelemetry.metrics.MeterProvider;
-import io.opentelemetry.metrics.spi.MeterProviderFactory;
-import io.opentelemetry.trace.DefaultTracerProvider;
 import io.opentelemetry.trace.Tracer;
 import io.opentelemetry.trace.TracerProvider;
-import io.opentelemetry.trace.propagation.HttpTraceContext;
-import io.opentelemetry.trace.spi.TracerProviderFactory;
-import java.util.Objects;
-import java.util.ServiceLoader;
-import javax.annotation.Nullable;
-import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * This class provides a static global accessor for telemetry objects {@link Tracer}, {@link Meter}
- * and {@link BaggageManager}.
+ * The entrypoint to telemetry functionality for tracing, metrics and baggage.
  *
- * <p>The telemetry objects are lazy-loaded singletons resolved via {@link ServiceLoader} mechanism.
+ * <p>A global singleton can be retrieved by {@link #get()}. The default for the returned {@link
+ * OpenTelemetry}, will include any {@link io.opentelemetry.trace.spi.TracerProviderFactory}, {@link
+ * io.opentelemetry.metrics.spi.MeterProviderFactory}, or {@link
+ * io.opentelemetry.baggage.spi.BaggageManagerFactory} found on the classpath, or otherwise will be
+ * default, with no-op behavior.
  *
  * @see TracerProvider
- * @see MeterProviderFactory
- * @see BaggageManagerFactory
+ * @see MeterProvider
+ * @see BaggageManager
  */
-@ThreadSafe
-public final class OpenTelemetry {
-  private static final Object mutex = new Object();
-
-  @Nullable private static volatile OpenTelemetry instance;
-
-  private final TracerProvider tracerProvider;
-  private final MeterProvider meterProvider;
-  private final BaggageManager contextManager;
-
-  private volatile ContextPropagators propagators = DefaultContextPropagators.builder().build();
+public interface OpenTelemetry {
 
   /**
-   * Returns a singleton {@link TracerProvider}.
+   * Returns the registered global {@link OpenTelemetry}, a default {@link OpenTelemetry} composed
+   * of functionality any {@link io.opentelemetry.trace.spi.TracerProviderFactory}, {@link
+   * io.opentelemetry.metrics.spi.MeterProviderFactory}, or {@link
+   * io.opentelemetry.baggage.spi.BaggageManagerFactory} found on the classpath, or otherwise will
+   * be default, with no-op behavior.
    *
-   * @return registered TracerProvider or default via {@link DefaultTracerProvider#getInstance()}.
-   * @throws IllegalStateException if a specified TracerProvider (via system properties) could not
-   *     be found.
+   * @throws IllegalStateException if a provider has been specified by system property using the
+   *     interface FQCN but the specified provider cannot be found.
    */
-  public static TracerProvider getTracerProvider() {
-    return getInstance().tracerProvider;
+  static OpenTelemetry get() {
+    return DefaultOpenTelemetry.getGlobalOpenTelemetry();
+  }
+
+  /** Returns the globally registered {@link TracerProvider}. */
+  static TracerProvider getGlobalTracerProvider() {
```

Comment:
Unfortunately the statics can't be named the same as the method name @carlosalberto. So added them with the "global" prefix for now.

Reply:
If I did this change, I would split the renaming in a separate PR, because that is a trivial change. Hard now to look at all changes and determine what is trivial vs what is behavior change.

Reply:
Do you mean making the instance methods something like getInstanceTracerProvider and swapping the names later? That seems like intentionally merging a bad name for a minor improvement in code review - the changes are only to this interface and the class. It's not something I'd generally recommend a contributor to have to do.

Reply:
Went ahead and renamed the instance methods, still don't think it's a good idea to merge this.